episodes/web-location-identifiers.Rmd

```diff
@@ -0,0 +1,257 @@
+6. To generalize, unique identification codes enable globally unique and persistent identification of a resource. These codes follow conventions in their representation and are registered centrally, enabling a uniform structure of these identifiers globally or in the user community. This makes unique identification codes unambiguous and referencable.
+7. (<i>Use this slide to collect unique identification codes in the research fields of the learners. We like to do this in a brief plenary discussion, asking the learners to share unique identifiers, they use regularly and what they represent.</i>)
+8. Now we will take this concept of uniform identifiers to the web.
+9. But first, we will need to define and distinguish between a few terms, that come up frequently in this context. The examples we have seen a couple of minutes ago, are - simply put - a series of structurally harmonized characters that uniquely identify a resource...
+10. The gene identifier we have seen, is basically just a synonym for the gene, which is actually called SULFITE REDUCTASE. In the same sense, your Orcid is just a name for a metadata record describing your academic personality. This is, what we call a URN, Uniform Resource **Name**. We know, what the resource is called, but the name gives no clue, on how we can retrieve the information stored behind this name.
+11. To do so on the web, we need the address...
+12. where we can find the resource with a specific name. The address alone is not enough, though. We also need information on how we can retrieve the resource...
+13. the communication protocol. We will focus on this technology in a minute. These information are given as a Uniform Resource Locator, URL: a series of characters, that specify the address of a resource and the protocol necessary to retrieve this resource.
+14. Both, URN and URL, can be used to identify a specific resource. You either need the name or the location, to identify data on the web. So URN and URL **are** Uniform Resource Identifier.
+15. We will further focus on URLs and the technology hidden behind this string of characters, that we enter to open a resource on the web.
+16. And for this we take the URL that retrieves the information for our gene example.
+17. So let's decode this character set: The resource we are looking for, is...
+18. hosted on a specific server. On this server, a certain file path needs to be followed, that will lead to the resource. The final resource might be stored in a data base structure. So the necessary query parameter are specified that lead to the data base entry for the requested resource. And don't forget, we will need to follow a specific protocol, in this case HTTPS, in order to communicate with the host server. HTTPS is an extension of the Hypertext Transfer Protocol, that uses encryption technology for secure communication on the network.
```

Comment:
```suggestion
18. hosted on a specific server. On this server, a certain file path needs to be followed, that will lead to the resource. The final resource might be stored in a database structure. So the necessary query parameter are specified that lead to the database entry for the requested resource. And don't forget, we will need to follow a specific protocol, in this case HTTPS, in order to communicate with the host server. HTTPS is an extension of the Hypertext Transfer Protocol, that uses encryption technology for secure communication on the network.
```

I would either not mix "file path" and "database" stuff, as it becomes factually wrong if you mix that.

You can explain fragments `#...` with file paths only, but then you have no good story for query parameters. I'd probably change the explanation to something like:

The resource we are looking for is hosted (NOTE: is *hosted* clear? otherwise say "provided" etc.) on a specific server. To explain which resource we want to get, the path in the URL gives a kind of "location address" to the server, so it knows where to find what we are looking for [...]

if you want to say something about query parameters: .... the allow to specify more details beyond stating the resource itself, e.g. "I want to order a coffee latte, but with soy milk and without sugar" - the server knows which item on the menu you want from the first part ("a caffee latte"), but the detailed requirements of how you want it to be served are often given in the query parameters - "with soy milk and without sugar"

I think the term "server" was actually originally picked because of this kind of metaphor, so I'd double-down on it.